src/Models/HydrostaticFreeSurfaceModels/split_explicit_free_surface.jl

```diff
@@ -245,9 +239,35 @@ struct ForwardBackwardScheme end
 end
 
 @inline cosine_averaging_kernel(τ::FT) where FT = τ >= 0.5 && τ <= 1.5 ? FT(1 + cos(2π * (τ - 1))) : zero(FT)
```

Comment:
is this called in a hot inner loop? It uses short-circuited logic but I don't think that's necessary here

when converting a number to some floating point, I think it's usually easier to read if you use

convert(FT, number)

rather than FT(long_expression).

I think the latter form should only be used for converting literals, as in `FT(1.5)`.

Reply:
It's for CPU use only so I guess it should be ok